lib/ansible/modules/user.py

```diff
@@ -1247,7 +1247,7 @@ def create_homedir(self, path):
             else:
                 skeleton = '/etc/skel'
 
-            if os.path.exists(skeleton):
+            if os.path.exists(skeleton) and os.path.isdir(skeleton):
```

Comment:
This change looks like it would just silently ignore `skeleton` if it wasn't a dir.  We effectively relied on the `OSError` to be the thing that errored if someone provided a file for example.

As such, this probably needs to add additional error checking.  Probably by means of also using `stat.S_ISCHR` to account for `/dev/null`

Reply:
Yes I see, skipping over in general is not a good idea if You have to rely on `OSError` if a file is provided.

`stat.S_ISCHR` is maybe a bit to general since it would return `True` also on other device-files, which is surely not welcome. Also, adding a lot of extra checking and error-handling just to make sure we just filter out `/dev/null` seems to be a bit to much for what we want to archive.

Maybe this suggestion takes us a bit closer? This way file-paths still bleed through and get caught while executing `shutil.copytree()`, also the intention is clearer:
```suggestion
            if os.path.exists(skeleton) and skeleton != os.devnull:
```

Reply:
I think the suggested change is ok and should account for sivel's concerns there.